econml/sklearn_extensions/model_selection.py

```diff
@@ -256,6 +270,536 @@ def get_n_splits(self, X, y, groups=None):
         return self.n_splits
 
 
+class ModelSelector(metaclass=abc.ABCMeta):
+    """
+    This class enables a two-stage fitting process, where first a model is selected
+    by calling `train` with `is_selecting=True`, and then the selected model is fit (presumably
```

Comment:
I am wondering why we opt for a 'train' method over a 'fit' method

Reply:
At first I used `fit`, but this makes the error messages much easier to understand if a selector is being passed where an estimator is expected or vice-versa.